packages/gestalt/src/InternalTextField.js

```diff
@@ -163,6 +165,7 @@ const InternalTextFieldWithForwardRef: React$AbstractComponent<Props, HTMLInputE
         // https://stackoverflow.com/questions/14447668/input-type-number-is-not-showing-a-number-keypad-on-ios
         pattern={type === 'number' ? '\\d*' : undefined}
         placeholder={placeholder}
+        readOnly={readonly}
```

Comment:
I was going to leave this comment below, but this is a perfect example: React thinks this should be camelCase, so I think we should too 😅